Add Ember test setup

unflxw opened the pull request:
### Initialise ember project

This commit creates a new ember project using the `ember new` command. You can ignore this commit.

### Set up Appsignal and add error route

Adds an `/error` route using `ember generate route` and adds Appsignal instrumentation to the project.

Also add required commands and Docker Compose setup for standard rack usage, and add it to the tests.

unflxw commented:
The tests are currently failing because of some issue with the `elixir/alpine-release` test setup; see #50.

luismiramirez commented on javascript/ember/app/.github/workflows/ci.yml:
```diff
@@ -0,0 +1,45 @@
+name: CI
```
Do we need this file?

unflxw replied:
Ah, no we don't. The `ember new` command adds a lot of stuff we don't really need. Thank you for noticing!

tombruijn commented:
I'm not getting any errors reported to AppSignal.com when I press the error link. Should I see errors being reported?

unflxw commented:
@tombruijn You should be seeing errors, yes! Like with the other front-end integrations (see e.g. [javascript/react](https://github.com/appsignal/test-setups/blob/main/javascript/react/app/src/Appsignal.js#L4)), the account-wide API key doesn't work -- you must write your API key in the application somehow, in this case, in `app/app.js`: https://github.com/appsignal/test-setups/pull/51/files#diff-316c7af1e017e3fb4426337375e4e4e134a8349794aec3f7cd0b1a2bb5853212R17

It is mentioned in the README, but nonetheless it's an unintuitive setup. This workflow should be improved at some point somehow, so it's as seamless as with the back-end apps. Some of the apps already have better workflows, using environment variables in some form ([vue](https://github.com/appsignal/test-setups/tree/main/javascript/vue-2#vue-2-cli-app-test-app), [vanilla](https://github.com/appsignal/test-setups/blob/2ca92c1f52ac322c65b4c88ebbf6a3816aeeb3a0/javascript/vanilla/README.md)) -- but none of them are consistent with each other, and the one-API-key-per-app bit of the problem remains unsolved.

tombruijn commented:
> the account-wide API key doesn't work -- you must write your API key in the application somehow, in this case, in app/app.js.

Yes, that didn't work for me. Assigning the `appsignal` object to `window` and calling `appsignal.demo()` does send some data. The error endpoint does not.

unflxw commented:
@tombruijn Found the issue, my bad! I initialised the ember application at `javascript/ember`, then realised that for consistency with the other projects it should be at `javascript/ember/app` instead, so I moved it and redid the commit history so that it had always been there.

When doing this history rewrite, I did not notice that the file for the error route added in the second commit ended up at `javascript/ember/app/routes`, when it should've ended up at `javascript/ember/app/app/routes` instead. Confusing!

To add to the confusion, Ember does not complain about the non-existing route, it just stays in the route it already is at -- which is _also_ what it does when it can't load a component in a template, which is the error that it's supposed to be triggering.

Anyway, moved files around, it should be in the right place now, and you should see an error about `this-does-not-exist` in prod shortly after visiting `/error`. Sorry about that!

tombruijn commented on javascript/ember/commands/run:
```diff
@@ -0,0 +1,20 @@
+  script/setup
+  mono bootstrap
+  mono clean
+  mono build
```
This takes forever on my laptop to build all packages. I suggest only building the packages we actually need.

```suggestion
  mono build --package=@appsignal/types,@appsignal/core,@appsignal/javascript,@appsignal/ember
```

unflxw replied:
I'll do a separate PR applying this change to this and other packages.

tombruijn reviewed (approved):
I see errors being reported now!
